src/functions-and-arrays.js

```diff
@@ -1,38 +1,156 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(a,b) 
+{
+  if (a>b) {
+    return a;
+    
+  } else {
+    return b;
+    
+  }
+}
+maxOfTwoNumbers(a,b);
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(wordsArr)
+ {
+  let longestWord = "";
+  if(wordsArr.length===0)
+  { 
+    return null;
+  } 
+  else if (wordsArr.length===1)
+  {
+     return wordsArr[0] ;
+  
+  }
+  else
+     for (let i = 0; i < wordsArr.length; i++) 
+     {
+    if ( wordsArr[i].length > longestWord.length) 
+    {
+
+      longestWord = wordsArr[i];
+    }
+  } 
+  return longestWord 
+}
+findLongestWord(wordsArr);
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) 
+
+ {
+  
+  var sum = 0;
+  if(numbers.length===0)
+  { 
+    return 0;
+  } 
+  else if (numbers.length===1)
+  {
+     return numbers[0] ;
+  
+  }
+  else 
+  {
+  for (var i = 0; i < numbers.length; i++) {
+      sum += numbers[i];
+  }
+  return sum;
+}
+
+    sumNumbers(numbers);
+ }
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(numbers) 
+{
+ let sum = 0;
+  if(numbers.length===0)
+  { 
+    return 0;
+  } 
+  else if (numbers.length===1)
+  {
+     return numbers[0] ;
+  
+  }
+else
+ for(let el of numbers)
+ {
+  switch(typeof el)
+  {
+      case 'number':
+      case 'boolean':
+          sum += el;
+        break;
+      case 'string':
+          sum += el.length;
+        break;
+      default:
+        throw new Error("unsupported data type(object or array)present in the array"); 
+      }
+ }
+ return sum;
+}
+   sum(numbers);
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbersAvg)
+ {var sum = 0;
+  var avg = 0;
+  let result = 0;
+  if (numbersAvg.length === 0) 
+  return null;
+  else if (numbersAvg.length === 1) 
+  {
+  return numbersAvg/numbersAvg.length;
+ }
+  const average =numbersAvg.reduce((a, b) => a + b, 0) / numbersAvg.length;
+    return avg;
+  for (var i = 0; i < numbersAvg.length; i++) 
```

Comment:
This solution would also be fine, but you should do the division after the loop is finished and divide by the amount of numbers in the array.